File: 2nd_Semester/SNP/example_code/praktika/snp_students/P04_Modularisieren_von_C_Code/work/modularize/java/triangle.java
/*****************************************************************************
    M. Thaler, Jan. 2000
    Datei:          triangle.java
    Funktion:       Die drei Seiten eines Dreiecks einlesen und bestimmen ob
                    das Dreieck rechtwinklig ist.
    Returns:        Nichts.
    Korrekturen:    - Maerz 2002, M. Thaler, H. Fierz
                      Abfrage bei unkorrekter Eingabe wiederholen
                    - Sept 2016, A. Gieriet
                      Refactored (sprechende Variablen-Namen,
                                  "magic" Numbers durch Symbole ersetzt, etc.)
******************************************************************************/

class triangle {
    
    public static void main(String[] args)
        throws java.io.IOException
    {
        int READ_ERROR = -2;
        int MAX_NUMBER = 1000;

        read ReadInt = new read();
        rectang Rect = new rectang();
        
        while (true) {
            System.out.println("\nDreiecksbestimmung (CTRL-C: Abbruch)\n");

            int word = 0;
            int a = 0;
            int b = 0;
            int c = 0;

            do {
                System.out.print("Seite a: ");
                word = ReadInt.getInt(MAX_NUMBER);
            }
            while ((word < 0) && (word != READ_ERROR));
            if (word >= 0)
                a = word;
            else
                break;
            
            do {
                System.out.print("Seite b: ");
                word = ReadInt.getInt(MAX_NUMBER);
            }
            while ((word < 0) && (word != READ_ERROR));
            if (word >= 0)
                b = word;
            else
                break;
            
            do {
                System.out.print("Seite c: ");
                word = ReadInt.getInt(MAX_NUMBER);
            }
            while ((word < 0) && (word != READ_ERROR));
            if (word >= 0)
                c = word;
            else
                break;
            
            if (Rect.Rectangular(a, b, c) == true)
                System.out.println("-> Dreieck " + a + "-" + b + "-" + c
                                   + " ist rechtwinklig");
            else
                System.out.println("-> Dreieck " + a + "-" + b + "-" + c
                                   + " ist nicht rechtwinklig");
            System.out.println("\n");
        }
        System.out.println("\n\nbye bye\n");
    }
}

File: 2nd_Semester/SNP/example_code/praktika/snp_students/P04_Modularisieren_von_C_Code/work/modularize/src/triangle.c
/* ----------------------------------------------------------------------------
 * --  _____       ______  _____                                              -
 * -- |_   _|     |  ____|/ ____|                                             -
 * --   | |  _ __ | |__  | (___    Institute of Embedded Systems              -
 * --   | | | '_ \|  __|  \___ \   Zuercher Hochschule Winterthur             -
 * --  _| |_| | | | |____ ____) |  (University of Applied Sciences)           -
 * -- |_____|_| |_|______|_____/   8401 Winterthur, Switzerland               -
 * ----------------------------------------------------------------------------
 */
/**
 * @file
 * @brief Lab implementation
 */
#include "read.h"
#include "rectang.h"
#include <stdio.h>
#include <stdlib.h>

/// max side length
#define MAX_NUMBER 1000

/**
 * @brief Main entry point.
 * @returns Returns EXIT_SUCCESS (=0) on success, EXIT_FAILURE (=1) on failure.
 */
int main(void) {
    // begin students to add code for task 4.1

    while (1) {
        printf("\nDreiecksbestimmung (CTRL-C: Abbruch)\n\n");

        int word = 0;
        int a = 0;
        int b = 0;
        int c = 0;

        do {
            printf("Seite a: ");
            word = getInt(MAX_NUMBER);
        } while ((word < 0) && (word != READ_ERROR));
        if (word >= 0)
            a = word;
        else
            break;

        do {
            printf("Seite b: ");
            word = getInt(MAX_NUMBER);
        } while ((word < 0) && (word != READ_ERROR));
        if (word >= 0)
            b = word;
        else
            break;

        do {
            printf("Seite c: ");
            word = getInt(MAX_NUMBER);
        } while ((word < 0) && (word != READ_ERROR));
        if (word >= 0)
            c = word;
        else
            break;

        if (isRectangle(a, b, c, 0) == 1)
            printf("-> Dreieck %d-%d-%d ist rechtwinklig\n\n", a, b, c);
        else
            printf("-> Dreieck %d-%d-%d ist nicht rechtwinklig\n\n", a, b, c);
        printf("\n");
    }
    printf("\n\nbye bye\n\n");
    // end students to add code
    return EXIT_SUCCESS;
}
